core/src/main/scala/org/http4s/Message.scala

```diff
@@ -211,8 +211,10 @@ final case class Request(
   def decodeWith[A](decoder: EntityDecoder[A], strict: Boolean)(f: A => Task[Response]): Task[Response] =
     decoder.decode(this, strict = strict).fold(_.toHttpResponse(httpVersion), f).join
 
-  override def toString: String =
-    s"""Request(method=$method, uri=$uri, headers=${headers})"""
+  override def toString: String = {
+    val newHeaders = headers.filter(_.is(Authorization)).filter(_.is(org.http4s.headers.Cookie))
```

Comment:
One thing I've done in some other clients is to replace the sensitive header value with `<REDACTED>`.  Then there's evidence of it being there, but it still doesn't leak.  Then it's a map to a raw header, and pass through non-matching ones as-is.

We might also want to use the `CaseInsenstiveString` getter, which will handle invalid values, instead of the `HeaderKey`.  That way we don't leak a credential just because someone forgot a quotation mark.